### PR TITLE
Build binaries that work on older macOS versions

### PR DESCRIPTION
**Pull request by brechtm**
See #16 

I managed to build a python executable on Sierra that ran on Yosemite using this Makefile. I experimented with this before my vacation, so unfortunately I don't remember the details. I'm not sure which `MACOSX_DEPLOYMENT_TARGET` I used. For another setting, the executable wouldn't run (anywhere IIRC).

Note that you need to have the appropriate SDK's installed. [xcodelegacy](/devernay/xcodelegacy) can help with that.

The build process seems to be failing now that I ran it today. I'll update this PR once I get it to build again.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -164,14 +180,19 @@ MACHINE_SIMPLE-$1=$$(MACHINE_SIMPLE-$$(ARCH-$1))
 else
 MACHINE_SIMPLE-$1=$$(ARCH-$1)
 endif
-SDK-$1=$$(basename $1)
+
+MACOSX_DEPLOYMENT_TARGET=10.6
```
Comment (brechtm) [inline Makefile]: With this set to 10.6, the build fails.

Setting it to 10.8, the build succeeds and I verified that the produced python binary runs on Yosemite (in a VirtualBox VM).

Comment (dimaqq) [inline Makefile]: IMO there's not point supporting 10.6... IIRC current XCode only works with 10.8 onwards anyway.

Comment (brechtm) [inline Makefile]: @dimaqq You need to install old SDK's manually in any case. See  [xcodelegacy](/devernay/xcodelegacy).

Comment (brechtm) [inline Makefile]: Fails with 10.7 as well:

```
...
Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libpython3.5m.a(pymath.o) has no symbols
xcrun --sdk macosx10.7 cc -arch x86_64 -mmacosx-version-min=10.7 --sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.7.sdk     -Wl,-stack_size,1000000  -framework CoreFoundation -o python.exe Programs/python.o libpython3.5m.a -ldl  -framework CoreFoundation
Running ld for x86_64 ...
./python.exe -E -S -m sysconfig --generate-posix-vars ;\
	if test $? -ne 0 ; then \
		echo "generate-posix-vars failed" ; \
		rm -f ./pybuilddir.txt ; \
		exit 1 ; \
	fi
/bin/sh: line 1: 94626 Killed: 9               ./python.exe -E -S -m sysconfig --generate-posix-vars
generate-posix-vars failed
make[1]: *** [pybuilddir.txt] Error 1
make: *** [build/macOS/Python-3.5.3-macosx.x86_64/dist/lib/libpython3.5m.a] Error 2
```

Comment (brechtm) [inline Makefile]: The python binary produced with `MACOSX_DEPLOYMENT_TARGET=10.8` also runs on Mountain Lion (10.8.5), verified in a VM.

**Comment by brechtm** [comment]
Just a friendly reminder to have another look at this PR (and #10) when you have the time... I'd love to be able to automatically produce macOS app packages on Travis using briefcase (which depends on the Python builds offered here).

Let me know if there's anything else I can do.

**Comment by phildini** [comment]
Oof! It sucks that we haven't gotten to this. Trying to prioritize it now.

**Comment by freakboy3742** [comment]
A ping on this issue - I attempted to build fresh packages over the last couple of weeks, and I've been hitting a wall... ironically, dealing with *recent* macOS builds. I've [recorded details here](https://gist.github.com/freakboy3742/2cdb1c68c189539d588ef766432b1d97) if you're able to shed any light on this.

**Comment by freakboy3742** [comment]
Another ping - it looks like the issue was a bug with Xcode 9; Xcode 9.1 has resolved the problem, and I'm in the process of updating the code to accomodate recent iOS and macOS releases, as well as setting a minimum OSX/iOS version so that older versions will be supported.

**Comment by freakboy3742** [comment]
Thanks for the patch; unfortunately, I needed to make a bunch of changes to get the builds working again, so I've integrated this patch as best I can manually. Binary uploads are happening now; should be done in an hour or so.

**Pull request closed without merging**